apps/web/src/app/(marketing)/blog/[slug]/page.tsx

```diff
@@ -0,0 +1,182 @@
+import type { Metadata } from "next";
+import Image from "next/image";
+import { notFound } from "next/navigation";
+
+import { Prose } from "@/components/prose";
+import { AnimatedGroup } from "@/components/ui/animated-group";
+import { getSinglePost } from "@/lib/blog-query";
+import {
+  delayedTransitionVariants,
+  transitionVariants,
+} from "@/lib/transitions";
+
```

Comment:
I don't often build static sites, but you can prerender the pages for each post at build time and then they'll become stale after 300 seconds. see https://nextjs.org/docs/app/guides/incremental-static-regeneration